plane/src/drone/mod.rs

```diff
@@ -82,6 +110,20 @@ pub async fn drone_loop(
                         tracing::error!(?err, "Error acking event.");
                     }
                 }
+                MessageToDrone::RenewKeyResponse(renew_key_response) => {
+                    let RenewKeyResponse { backend, deadlines } = renew_key_response;
+                    tracing::info!(?deadlines, "Received key renewal response.");
+
+                    if let Some(deadlines) = deadlines {
+                        key_manager
+                            .lock()
+                            .expect("Key manager lock poisoned.")
+                            .update_deadlines(&backend, deadlines);
+                    } else {
+                        // TODO: we could begin the graceful termiation here.
```

Comment:
nit: misspelling

Reply:
Also, we need to implement this before the implementation is actually correct, right?

Reply:
Oh, nevermind, I see that this happens in the renew key loop